lib/metadata/metadataUtils.js

```diff
@@ -7,16 +7,29 @@ const { isBucketAuthorized, isObjAuthorized } =
     require('../api/apiUtils/authorization/permissionChecks');
 const bucketShield = require('../api/apiUtils/bucket/bucketShield');
 
+/**
+  * Compute and add an isNonversionedBucket flag to options
+  * @param {object} options - the current options
+  * @param {object} bucketMD - bucketMD object
+  * @return {object} - the updated options objects
+  */
+function addIsNonversionedBucket(options, bucketMD) {
+    const isNonversionedBucket = bucketMD
```

Comment:
I suggest to only add the option when it's true, and not add it otherwise (to avoid any change when unnecessary and slightly reduce the payload to parse server-side)